### PR TITLE
Add QR Login Remote Service

### PR DESCRIPTION
**Pull request by emilylaguna**
WPiOS Related PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/18756

### Description
Adds the QR Login Remote Service to validate and authenticate the token/data coming from a QR Code

### Testing Details

Test using the steps from the related PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/18756 

- [x] Please check here if your pull request includes additional test coverage.
- [x] I have considered updating the `version` in the `.podspec` file.

### CODE REVIEW

File: WordPressKit/QR Login/QRLoginServiceRemote.swift
Comment (momo-ozawa) [inline WordPressKit/QR Login/QRLoginServiceRemote.swift]: [nit]
```suggestion
    public func validate(token: String, data: String, success: @escaping (QRLoginValidationResponse) -> Void, failure: @escaping (Error?, QRLoginError?) -> Void) {
```

File: WordPressKit.podspec
Comment (momo-ozawa) [inline WordPressKit.podspec]: The podspec version will probably be out of date by the time you merge this 👀

**Review by momo-ozawa — COMMENTED** [review]
@emilylaguna 

~Corresponding WPiOS PR reviewed [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/18756#pullrequestreview-990899817). I'll re-review this once the issue with the QR codes is resolved.~

WPiOS PR re-reviewed [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/18756#issuecomment-1152378982).

Also, are you planning on adding tests for the new endpoints?

**Comment by emilylaguna** [comment]
> Also, are you planning on adding tests for the new endpoints?
Yeah, I'm going to add testing for all the newly added code.

**Comment by emilylaguna** [comment]
> The podspec version will probably be out of date by the time you merge this 👀

True. I'll update it before I merge in

**Review by momo-ozawa — APPROVED** [review]

**Comment by emilylaguna** [comment]
@momo-ozawa may I get another review? I added the tests in finally.

**Review by momo-ozawa — APPROVED** [review]
LGTM, tests included! 🎉 
Looks like the podspec version is okay too (for now at least) ☺️

**Pull request merged**
